File: length-of-last-word/length-of-last-word.cpp
#include <iostream>
#include <string>
using namespace std;


//       Speed  Memory
//Total  0ms    6.5MB    
//Beats  100%   92.69%
//




int lengthOfLastWord(string s) {
    int len = 0;

    //Trim from the end of the string any white space
    while(s[s.size() - 1] == ' '){
        s.pop_back();
    }
    for(int i = s.size() - 1; i >= 0 ; --i){

        if(s[i] != ' ' ){
            len += 1;
        }
        else{
            return len;
        }
    }

    return len;
}




int main(){
    cout << "Type words to find length of last word in list: ";
    string word = "";
    cin >> word;
    cout << lengthOfLastWord(word) << endl;
}

File: length-of-last-word/length-of-last-word.dart
import 'dart:io';

//This is my first dart program!!!
//I am doing this to start my learning process of flutter
//Runtime: 249ms Beats: 78.21%
//Memory: 140.9 Beats: 42.46%
void main(){
    Solution sol = new Solution();
    stdout.write("Enter words to find length of last word: ");
    String input_str = stdin.readLineSync()!;
    print(sol.lengthOfLastWord(input_str));
}


class Solution {
    int lengthOfLastWord(String s) {
    //Clear the spaces from the end of the string
      s = s.trim();
      int len = 0;
      for(int i = s.length - 1 ; i >= 0; --i){
        if(s[i] == " "){
          break;
        }
        else{
          len += 1;
        }
      }
      return len;
  }
}


